### DISCUSSION TITLE
Add support for configurable and dynamic online status, not always online

### DISCUSSION THREAD

**Discussion by samtrek** [d99kris/nchat#122] [Ideas]
How does nchat handle offline and online status to other normal whatsapp users? Seems my own nchat whatsapp is always online, which is a function that I dont desire, I have looked through the options to see if I can see an option to turn it off, I have not seen any, is there anyway not to show always online to other users?

**Comment by d99kris** [comment]
Hi @samtrek - yep, currently `nchat` presents the user as online while `nchat` is running. 
Would you like to see an option to disable that completely? 
Or are you looking for `nchat` to detect user activity and show online status based on that?

**Comment by samtrek** [comment]
Yes I will like to see an option to in this order
1. completely disable online status; this can be in form of a toggle, like true or false
2. Update status on interaction with nchat; this is for those that want to be shown online when they are replying to their chat, and nchat to revert back to offline when the interaction is over
Left to me I will prefer option 1. Because I leave nchat open in the background when I am doing other things and only interact with it when there are new messages. So been seen online all the time when I am sleeping can be inconvenient  at times, as people keep on asking why have not been answering my messages when they can see that I am online.

**Reply by d99kris** [reply]
Thanks for elaborating on the suggested functionality! I think it makes sense and I will look into it when I find some time.

**Reply by samtrek** [reply]
Thank you for your time, and for the efforts you put in to making the app.

**Reply by d99kris** [reply]
Thanks! FYI - functionality to control whether to share online (and typing) status has been added now, as well as an option to have online status follow whether terminal is active or not. See #127 for the new config parameter names.

**Reply by samtrek** [reply]
I have tested the new option you included in #127 and it is working quite well, thank you for the update, do really appreciate it, now my default desktop whatsapp client.
